server/words.js

```diff
@@ -8,53 +8,57 @@ const easy = ['cat', 'dog', 'umbrella', 'mouse', 'apple', 'banana', 'pineapple']
 const medium = ['merry-go-round', 'polaroid', 'photo album']
 const hard = ['high school', 'carnival', 'castle']
 const impossible = ['mystery', 'orange', 'novel', 'regret', 'excited']
-const custom = {}
+// const custom = {}
 
-const words = { easy, medium, hard, impossible, custom }
+const DEFAULT_WORDS = { easy, medium, hard, impossible, custom: [] }
+
+const wordSets = {}
 
 function addCustomWord(roomCode, word) {
-  if (Object.prototype.hasOwnProperty.call(custom, roomCode)) {
-    if (!custom[roomCode].find((current) => current === word)) {
-      custom[roomCode] = [...custom[roomCode], word]
-    }
-    console.log('if room already exists', Object.values(custom))
-  } else {
-    custom[roomCode] = [word]
-    console.log('room doesnt exists', Object.values(custom))
+  if (!wordSets[roomCode].custom.includes(word)) {
```

Comment:
I think if the roomCode doesn't exist in wordSets, its going to throw an error because you'd be trying to access the value custom from undefined. I think you need to do an outer check first to see if the roomCode exists in wordSets. 
```javascript
if (Object.prototype.hasOwnProperty.call(wordSets, roomCode)){
      if (!wordSets[roomCode].custom.includes(word)) {
            wordSets[roomCode].custom = [...wordSets[roomCode].custom, word]
     }
}  
```

Reply:
though now that i look at the code more it's pretty unlikely that the roomCode wont exist but feel free to add it if you'd like to have an extra safety check lol